python/lsst/ctrl/bps/wms/panda/panda_service.py

```diff
@@ -198,38 +201,38 @@ def from_generic_workflow(cls, config, generic_workflow, out_prefix, service_cla
         idds_workflow = cls(generic_workflow.name, config)
         workflow_generator = IDDSWorkflowGenerator(generic_workflow, config)
         idds_workflow.generated_tasks = workflow_generator.define_tasks()
-        cloud_prefix = config['bucket'] + '/' + \
+        file_placement_path = config['fileDistributionEndPoint'] + '/' + \
             config['payload_folder'] + '/' + config['workflowName'] + '/'
-        cls.copy_pickles_into_cloud([config['bps_defined']['run_qgraph_file']], cloud_prefix)
+        cls.copy_files_for_distribution([config['bps_defined']['run_qgraph_file']], file_placement_path)
         _LOG.debug("panda dag attribs %s", generic_workflow.run_attrs)
         return idds_workflow
 
     @staticmethod
-    def copy_pickles_into_cloud(local_pfns, cloud_prefix):
+    def copy_files_for_distribution(local_pfns, file_placement_path):
         """
         Brings locally generated pickle files into Cloud for further utilization them on the egde nodes.
         Parameters
         ----------
         local_pfns: `str`
             Local path to the file to be copied
 
-        cloud_prefix: `str`
-            Path on the cloud, including access protocol, bucket name to place files
+        file_placement_path: `str`
+            Path on the edge node accessed storage, including access protocol, bucket name to place files
         """
 
         copy_executor = concurrent.futures.ThreadPoolExecutor(max_workers=10)
         future_file_copy = []
         for src_path in local_pfns:
             src = ButlerURI(src_path)
-            target_base_uri = ButlerURI(cloud_prefix)
+            target_base_uri = ButlerURI(file_placement_path)
 
             # S3 clients explicitly instantiate here to overpass this
             # https://stackoverflow.com/questions/52820971/is-boto3-client-thread-safe
             target_base_uri.exists()
             src.exists()
 
             target = target_base_uri.join(os.path.basename(src_path))
```

Comment:
don't use os.path.basename if you are using `ButlerURI` already. Use `target_base_uri.join(src.basename())`